src/ai/backend/manager/api/resource.py

```diff
@@ -601,29 +601,27 @@ async def get_time_binned_monthly_stats(request: web.Request, user_uuid=None):
         io_read_bytes = 0
         io_write_bytes = 0
         disk_used = 0
+        idx = 0
         # Accumulate stats for containers overlapping with this time window.
-        while (
-            idx < rowcount
-            and ts + time_window > rows[idx].created_at.timestamp()
-            and ts < rows[idx].terminated_at.timestamp()
-        ):
+        while idx < rowcount and ts + time_window > rows[idx].created_at.timestamp():
             # Accumulate stats for overlapping containers in this time window.
-            row = rows[idx]
-            num_sessions += 1
-            cpu_allocated += int(row.occupied_slots.get("cpu", 0))
-            mem_allocated += int(row.occupied_slots.get("mem", 0))
-            if "cuda.devices" in row.occupied_slots:
-                gpu_allocated += int(row.occupied_slots["cuda.devices"])
-            if "cuda.shares" in row.occupied_slots:
-                gpu_allocated += Decimal(row.occupied_slots["cuda.shares"])
-            raw_stat = await redis_helper.execute(
-                root_ctx.redis_stat, lambda r: r.get(str(row["id"]))
-            )
-            if raw_stat:
-                last_stat = msgpack.unpackb(raw_stat)
-                io_read_bytes += int(nmget(last_stat, "io_read.current", 0))
-                io_write_bytes += int(nmget(last_stat, "io_write.current", 0))
-                disk_used += int(nmget(last_stat, "io_scratch_size/stats.max", 0, "/"))
+            if ts < rows[idx].terminated_at.timestamp():
+                row = rows[idx]
+                num_sessions += 1
+                cpu_allocated += int(row.occupied_slots.get("cpu", 0))
+                mem_allocated += int(row.occupied_slots.get("mem", 0))
+                if "cuda.devices" in row.occupied_slots:
+                    gpu_allocated += int(row.occupied_slots["cuda.devices"])
+                if "cuda.shares" in row.occupied_slots:
+                    gpu_allocated += Decimal(row.occupied_slots["cuda.shares"])
+                raw_stat = await redis_helper.execute(
+                    root_ctx.redis_stat, lambda r: r.get(str(row["id"]))
+                )
+                if raw_stat:
+                    last_stat = msgpack.unpackb(raw_stat)
+                    io_read_bytes += int(nmget(last_stat, "io_read.current", 0))
+                    io_write_bytes += int(nmget(last_stat, "io_write.current", 0))
+                    disk_used += int(nmget(last_stat, "io_scratch_size/stats.max", 0, "/"))
```

Comment:
```suggestion
        for row in rows:
            if ts + time_window <= row.created_at.timestamp() or ts >= row.terminated_at.timestamp():
                continue
            num_sessions += 1
            cpu_allocated += int(row.occupied_slots.get("cpu", 0))
            mem_allocated += int(row.occupied_slots.get("mem", 0))
            if "cuda.devices" in row.occupied_slots:
                gpu_allocated += int(row.occupied_slots["cuda.devices"])
            if "cuda.shares" in row.occupied_slots:
                gpu_allocated += Decimal(row.occupied_slots["cuda.shares"])
            raw_stat = await redis_helper.execute(
                root_ctx.redis_stat, lambda r: r.get(str(row["id"]))
            )
            if raw_stat:
                last_stat = msgpack.unpackb(raw_stat)
                io_read_bytes += int(nmget(last_stat, "io_read.current", 0))
                io_write_bytes += int(nmget(last_stat, "io_write.current", 0))
                disk_used += int(nmget(last_stat, "io_scratch_size/stats.max", 0, "/"))
```
We can simply iterate the rows.

Reply:
reflected! thanks :)